src/borg/repository.py

```diff
@@ -1475,6 +1476,25 @@ def delete_segment(self, segment):
         except FileNotFoundError:
             pass
 
+    def clear_empty_dirs(self):
+        """Delete empty segment dirs, i.e those with no segment files.
+        """
+        data_dir = os.path.join(self.path, 'data')
+        segment_dirs = (f.path for f in os.scandir(data_dir) if f.is_dir() and f.name.isdigit())
+        for segment_dir in segment_dirs:
+            try:
+                # os.rmdir will only delete the directory if it is empty
+                # so we don't need to explicitly check for emptiness first.
+                os.rmdir(segment_dir)
+            except FileNotFoundError:
+                # This may be indication of another parallel process cleaning up the dir, so log it.
+                logger.debug("While trying to clear empty segments dir, it seems it already vanished. "
+                             "Is another process doing cleanup?.")
```

Comment:
typo (remove the dot at the end, you already have a question mark).

Reply:
also, the repo is locked while borg operates in it, so there can be no parallel process.

so you could just remove 1489..1492 and nothing would be lost. just add a note about "file not found" case in the OSError handler.